amd/src/app/block/element.js

```diff
@@ -208,6 +209,70 @@ export default class BlockElement {
         });
     }
 
+    setupBulkEdit() {
+        const selectAllCheckbox = this.#element.querySelector('#select_all_box');
+        const selectAllLabel = this.#element.querySelector('#select_all_label');
+        const bulkDeleteButton = this.#element.querySelector('#block_sharing_cart_bulk_delete_confirm');
+        const cancelBulkDeleteButton = this.#element.querySelector('#block_sharing_cart_cancel_bulk_delete');
+        const bulkDeleteTrigger = this.#element.querySelector('#block_sharing_cart_bulk_delete');
+
+        const checkboxSelector = '.sharing_cart_item input[data-action="bulk_select"][type="checkbox"]';
+        const itemCheckboxes = this.#element.querySelectorAll(checkboxSelector);
```

Comment:
```suggestion
```
This shouldn't be a saved variable. By doing so you don't take new or deleted elements into account when selecting/deselecting.

Fixing this can be done by fetching the checkbox items every time an action or calculation are based upon them.

To test this try adding an element to the sharing cart and without refreshing the page use the select/deselect all functionality. You should see it won't change the checkbox of the new item